alphacsc/init_dict.py

```diff
@@ -24,7 +24,7 @@
 tied = custom_distances.translation_invariant_euclidean_distances
 
 
-class BaseDictGenerator():
+class BaseDictionary():
 
     def __init__(self, n_channels, n_atoms, n_times_atom, random_state,
                  window):
```

Comment:
I would pass init parameters here `D_init` and `D_init_params` so we simply pass `X` in the `init` method.

Reply:
As the `d_solver` is created at the same time as the initialization, I would say it is ok to pass `D_init` and `D_init_params` directly in the constructor.

The rational for this is the following: `D_init` and `D_init_params` are parameters of the `d_solver` so following the `sklearn` API, they should be passed in the constructor. `X` is data related so it should be passed in the `fit` method, which to me correspond to the `initialize_dictionary` method.